common-service/src/main/java/org/sunbird/scheduler/SchedulerManager.java

```diff
@@ -0,0 +1,144 @@
+      ProjectLogger.log("SchedulerManager:initScheduler: Error in starting scheduler jobs.", e);
+    } finally {
+      registerShutDownHook();
+    }
+    ProjectLogger.log("SchedulerManager:initScheduler: started scheduler jobs.");
+  }
+
+  /** This method will register the quartz scheduler job. */
+  public abstract void schedule();
+
+  /**
+   * Method to read the system properties and load to the Properties file that will be used to setup
+   * the quartz scheduler in distributed mode
+   *
+   * @return Properties file contains key(attributes) and values related to Postgres database
+   * @throws IOException
+   */
+  private Properties setUpClusterMode() throws IOException {
+    Properties configProp = new Properties();
+    InputStream in = this.getClass().getClassLoader().getResourceAsStream(FILE);
+    String host = System.getenv(JsonKey.SUNBIRD_PG_HOST);
+    String port = System.getenv(JsonKey.SUNBIRD_PG_PORT);
+    String db = System.getenv(JsonKey.SUNBIRD_PG_DB);
+    String username = System.getenv(JsonKey.SUNBIRD_PG_USER);
+    String password = System.getenv(JsonKey.SUNBIRD_PG_PASSWORD);
+    ProjectLogger.log(
+        "SchedulerManager:setUpClusterMode: Environment variable value for PostGress SQl= host, port,db,username,password "
+            + host
+            + " ,"
+            + port
+            + ","
+            + db
+            + ","
+            + username
+            + ","
+            + password,
+        LoggerEnum.INFO.name());
+    if (!StringUtils.isBlank(host)
+        && !StringUtils.isBlank(port)
+        && !StringUtils.isBlank(db)
+        && !StringUtils.isBlank(username)
+        && !StringUtils.isBlank(password)) {
+      ProjectLogger.log(
+          "SchedulerManager:setUpClusterMode: Taking Postgres value from Environment variable...",
+          LoggerEnum.INFO.name());
+      configProp.load(in);
+      configProp.put(
+          "org.quartz.dataSource.MySqlDS.URL", "jdbc:postgresql://" + host + ":" + port + "/" + db);
+      configProp.put("org.quartz.dataSource.MySqlDS.user", username);
+      configProp.put("org.quartz.dataSource.MySqlDS.password", password);
+    } else {
+      ProjectLogger.log(
+          "SchedulerManager:setUpClusterMode: Environment variable is not set for postgres SQl.",
+          LoggerEnum.INFO.name());
+      configProp = null;
+    }
+    return configProp;
+  }
+
+  /**
+   * Method to get the instance of the scheduler manager
+   *
+   * @return SchedulerManager instance
+   */
+  public abstract SchedulerManager getInstance();
+
+  /**
+   * This class will be called by registerShutDownHook to register the call inside jvm , when jvm
+   * terminate it will call the run method to clean up the resource.
+   *
+   * @author Manzarul
+   */
+  class ResourceCleanUp extends Thread {
+    @Override
+    public void run() {
+      ProjectLogger.log(
+          "SchedulerManager:ResourceCleanUp: started resource cleanup for Quartz job.");
```

Comment:
"SchedulerManager:ResourceCleanUp: Started resource cleanup for quartz scheduler."

"SchedulerManager:ResourceCleanUp: Generic exception while shutting down quartz scheduler = " + e.getMessage()

"SchedulerManager:ResourceCleanUp: Completed resource cleanup for quartz scheduler."